compiler/rustc_hir_analysis/src/collect/predicates_of.rs

```diff
@@ -726,14 +706,45 @@ pub(super) fn assert_only_contains_predicates_from<'tcx>(
                     ty::ClauseKind::RegionOutlives(_)
                     | ty::ClauseKind::ConstArgHasType(_, _)
                     | ty::ClauseKind::WellFormed(_)
-                    | ty::ClauseKind::ConstEvaluatable(_) => {
+                    | ty::ClauseKind::ConstEvaluatable(_)
+                    | ty::ClauseKind::HostEffect(..) => {
                         bug!(
                             "unexpected non-`Self` predicate when computing `{filter:?}` implied bounds: {clause:?}"
                         );
                     }
                 }
             }
         }
+        PredicateFilter::ConstIfConst => {
+            for (clause, _) in bounds {
+                match clause.kind().skip_binder() {
+                    ty::ClauseKind::HostEffect(_) => {}
```

Comment:
could probably check that the predicate indeed has `HostPolarity::Maybe` here also.